src/compiler/commandLineParser.ts

```diff
@@ -371,13 +371,30 @@ namespace ts {
                     parseResponseFile(s.slice(1));
                 }
                 else if (s.charCodeAt(0) === CharacterCodes.minus) {
-                    s = s.slice(s.charCodeAt(1) === CharacterCodes.minus ? 2 : 1).toLowerCase();
+                    s = s.slice(s.charCodeAt(1) === CharacterCodes.minus ? 2 : 1);
 
                     // Try to translate short option names to their full equivalents.
-                    if (hasProperty(shortOptionNames, s)) {
-                        s = shortOptionNames[s];
+                    if (hasProperty(shortOptionNames, s.toLowerCase())) {
+                        s = shortOptionNames[s.toLowerCase()];
+                    }
+                    else {
+                        // When using long-form switches, we follow standard command-line conventions and accept
+                        // "--example=VALUE", but we also accept "--example VALUE".
+                        const longFormSwitch = s.split("=", 1).toString().toLowerCase();
+                        if (longFormSwitch.length < s.length && hasProperty(optionNameMap, longFormSwitch)) {
+                            // It's in "--example=VALUE" format.  Replace it in the arg list with the separated format.
+                            const value = s.substring(longFormSwitch.length + 1);
+                            const opt = optionNameMap[longFormSwitch];
+                            if (opt.type === "boolean") {
+                              errors.push(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_with_option_1, opt.name, value));
+                              continue;
+                            }
+                            args.splice(i - 1, 1, opt.name, value);

```

Comment:
Don't modify the argument list, the caller hasn't necessarily expected that - but I'd rather you not copy the entire array to achieve this either.